python/hsfs/core/vector_server.py

```diff
@@ -337,11 +500,373 @@ def get_inference_helpers(
+
+        Currently timestamp which are in the database are returned as string. Whereas
+        passed features which were given as datetime are returned as integer timestamp.
+
+        # Arguments:
+            timestamp_value: The timestamp value to be handled, either as int or str.
+        """
+        if isinstance(timestamp_value, int):
+            _logger.debug(
+                f"Converting timestamp {timestamp_value} to datetime from UNIX timestamp."
+            )
+            return datetime.fromtimestamp(
+                timestamp_value / 1000, tz=timezone.utc
+            ).replace(tzinfo=None)
+        elif isinstance(timestamp_value, str):
+            _logger.debug(
+                f"Parsing timestamp {timestamp_value} to datetime from SQL timestamp string."
+            )
+            return datetime.strptime(timestamp_value, self.SQL_TIMESTAMP_STRING_FORMAT)
+        elif isinstance(timestamp_value, datetime):
+            _logger.debug(
+                f"Returning passed timestamp {timestamp_value} as it is already a datetime."
+            )
+            return timestamp_value
+        else:
+            raise ValueError(
+                f"Timestamp value {timestamp_value} was expected to be of type int or str."
+            )
+
+    def set_return_feature_value_handlers(
+        self, features: List[tdf_mod.TrainingDatasetFeature]
+    ) -> List[Callable]:
+        """Build a dictionary of functions to convert/deserialize/transform the feature values returned from RonDB Server.
+
+        TODO: Trim down logging once stabilised
+
+        Re-using the current logic from the vector server means that we currently iterate over the feature vectors
+        and values multiple times, as well as converting the feature values to a dictionary and then back to a list.
+
+        - The handlers do not need to handle the case where the feature value is None.
+        - The handlers should first convert/deserialize the feature values and then transform them if necessary.
+        - The handlers should be reusable without paying overhead for rebuilding them for each feature vector.
+        """
+        self._return_feature_value_handlers = {}
+        self._ordered_feature_group_feature_names = []
+        _logger.debug(
+            f"Setting return feature value handlers for Feature View {self._feature_view_name},"
+            f" version: {self._feature_view_version} in Feature Store {self._feature_store_name}."
+        )
+        for feature in features:
+            # These features are not part of the feature vector.
+            if feature.label or feature.training_helper_column:
+                _logger.debug(
+                    f"Skipping Feature {feature.name} as it is a label or training helper column."
+                )
+                continue
+
+            self._ordered_feature_group_feature_names.append(
+                feature.feature_group_feature_name
+            )
+            if (
```

Comment:
I think applying transformations should be another step after readign the data because as part of Model Dependent Transformation function. A transformation funciton is not realted to a single features but can require multiple features as input. So I think the _apply_transformation_function() should be retained. 

But I can make that change when I rebase it for my PR.